### PR TITLE
Added expansion panels demo to gallery.

### PR DESCRIPTION
**Pull request by dragostis**

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Given where we are in our demo cycle, please get @sethladd and/or @HansMuller to sign off before we add this to the default list.  We have several demos which are not on the default list like calculator, minesweeper and datatables.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+enum _Location {
+  Barbados,
+  Bahamas,
+  Bermuda
+}
+
+typedef String StringValue();
+
+class _ExpansionPanelDemoState extends State<ExpasionPanelsDemo> {
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: The convention is to give this class the same name as the StatefulWidget plus "State", so _ExpansionPanelsDemoState.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+import 'package:flutter/material.dart';
+
+class ExpasionPanelsDemo extends StatefulWidget {
+  static const String routeName = '/expansion_panels';
+
+  @override
+  _ExpansionPanelDemoState createState() => new _ExpansionPanelDemoState();
+}
+
+enum _Location {
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: Another convention: we usually put enums like this before the class defs.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  ExpansionPanelHeaderBuilder _createHeader(StringValue value, String name, String hint) {
+    return (BuildContext context, bool isExpanded) {
+      return new Row(
+        children: <Widget>[
+          new Flexible(
+            flex: 2,
+            child: new Container(
+              margin: const EdgeInsets.only(left: 24.0),
+              child: new DefaultTextStyle(
+                style: new TextStyle(
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: const TextStyle

**Comment by HansMuller** [comment]
This is a nice demo (I've seen it!) but it could be better example.  Maybe take another pass and add a little structure: some widget classes and one small value (like Vacation) class.  The result will probably be longer and hopefully more copy-and-paste worthy.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+        new Flexible(
+          flex: 2,
+          child: new Container(
+            margin: const EdgeInsets.only(left: 24.0),
+            child: new DefaultTextStyle(
+              style: const TextStyle(
+                color: Colors.black87,
+                fontSize: 15.0
+              ),
+              child: new Text(name)
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: You don't need a inherited widget (DefaultTextStyle) here, you can just specify the Text style: parameter.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+            )
+          )
+        ),
+        new Flexible(
+          flex: 3,
+          child: new Container(
+            margin: const EdgeInsets.only(left: 24.0),
+            child: new DefaultTextStyle(
+              style: const TextStyle(
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: In this case, and the previous one, defining class variables for the text styles and specifying them explicitly would be a little simpler. Using DefaultTextStyle is a good option when the widget's descendants come from elsewhere.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+            margin: const EdgeInsets.only(
+              left: 24.0,
+              right: 24.0,
+              bottom: 24.0
+            ) - margin,
+            child: new Center(
+              child: child
+            )
+          ),
+          new Container(
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: You could use a Divider here.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
+            ),
+            child: new Row(
+              mainAxisAlignment: MainAxisAlignment.end,
+              children: <Widget>[
+                new Container(
+                  margin: const EdgeInsets.only(right: 8.0),
+                  child: new FlatButton(
+                    onPressed: onCancel,
+                    child: new DefaultTextStyle(
+                      style: new TextStyle(
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: Same comment as earlier

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
+  });
+
+  final EdgeInsets margin;
+  final Widget child;
+  final VoidCallback onSave;
+  final VoidCallback onCancel;
+
+  @override
+  Widget build(BuildContext context) {
+    return new DefaultTextStyle(
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: I would wrap the Center's child with this, define the style as a class variable, and make the SAVE button's style explicit.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+  String get location => _location.toString().split(".")[1];
+  double _sunAmount = 80.0;
+  String get sunAmount => '${_sunAmount.round()}';
+  List<bool> expanded = new List<bool>.filled(3, false);
+
+  Widget _body(Widget child, int index, {EdgeInsets margin: EdgeInsets.zero}) {
+    return new CollapsibleBody(
+      margin: margin,
+      child: child,
+      onSave: () {
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: This sort of thing is often factored out as a _close() method.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
+  }
+
+  void _changeLocation(_Location newLocation) {
+    setState(() {
+      _location = newLocation;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: The way this is constructed could be less fragile. Imagine inserting another expansion panel or deleting. The expanded array and its indices need to be updated in many places. 

It seems like an aspect of the list of panels, probably in terms of their expanded state and the contents of their headers, should be factored out.

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 11]
```diff
@@ -0,0 +1,319 @@
...
+  final String value;
+  final String hint;
+  final bool showHint;
+
+  Curve _flipCurves(bool isExpanded) {
+    return isExpanded ? new Interval(0.0, 0.6, curve: Curves.ease) :
+        new Interval(0.4, 1.0, curve: Curves.ease);
+  }
+
+  Widget _crossFade(Widget first, Widget second, bool isExpanded) {
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: don't we have a widget for that now?

Comment (dragostis) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: @Hixie I asked myself the same question.

The widget is private, so you don't have access to it.

Comment (Hixie) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: Maybe we should make it public. :-)

File: examples/flutter_gallery/lib/demo/expansion_panels_demo.dart [hunk 12]
```diff
@@ -0,0 +1,335 @@
...
+          child: new Container(
+            margin: const EdgeInsets.only(left: 24.0),
+            child: new Text(name, style: text87)
+          )
+        ),
+        new Flexible(
+          flex: 3,
+          child: new Container(
+            margin: const EdgeInsets.only(left: 24.0),
+            child: _crossFade(
```
Comment (HansMuller) [inline examples/flutter_gallery/lib/demo/expansion_panels_demo.dart]: We don't usually call a function for a case like this, just inline the AnimatedCrossFade

**Comment by HansMuller** [comment]
LGTM

**Pull request merged**